atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+            while (resultSet.next()) {
+                tableNames.add(resultSet.getString("table_name"));
+            }
+            return tableNames.build();
+        });
+    }
+
+    private Connection applyPragma(Connection connection) throws SQLException {
+        // Oracle likes are case-sensitive. This pragma ensures that we mimic that behaviour with our sqlite queries
+        connection.createStatement().executeUpdate("PRAGMA case_sensitive_like = true;");
+        return connection;
+    }
+
+    private String getConnectionUrl() {
+        return "jdbc:sqlite:" + sqliteFileLocation.getAbsolutePath();
+    }
+
+    private String generatePrefixedTableName(String prefix, TableReference tableReference) {
+        OracleDdlConfig config = ImmutableOracleDdlConfig.builder()
+                .tablePrefix(prefix)
+                .overflowMigrationState(OverflowMigrationState.UNSTARTED)
+                .build();
+
+        OracleTableNameGetter tableNameGetter = OracleTableNameGetter.createDefault(config);
+        return tableNameGetter.getPrefixedTableName(tableReference);
+    }
+
+    /**
+     * A lightweight wrapper around {@link SimpleTimedSqlConnectionSupplier} that marshals some queries containing
+     * Oracle specific keywords such as PURGE to sqlite compatible queries
+     */
+    private static class SqliteSqlConnectionSupplier implements SqlConnectionSupplier {
+        private final SqlConnectionSupplier delegate;
+
+        SqliteSqlConnectionSupplier(ConnectionSupplier delegate) {
+            this.delegate = new SimpleTimedSqlConnectionSupplier(delegate);
+        }
+
+        @Override
+        public SqlConnection get() {
+            return marshalOracleQueryToSqlite(delegate.get());
+        }
+
+        @Override
+        public void close() throws PalantirSqlException {
+            delegate.close();
+        }
+
+        private SqlConnection marshalOracleQueryToSqlite(SqlConnection sqlConnection) {
+            SqlConnection spy = spy(sqlConnection);
+            doAnswer(invocation -> {
+                        String sql = invocation.getArgument(0);
+                        String strippedSql = StringUtils.removeEnd(sql, "PURGE");
+                        sqlConnection.executeUnregisteredQuery(strippedSql);
+                        return null;
+                    })
+                    .when(spy)
+                    .executeUnregisteredQuery(endsWith("PURGE"), any());
+            return spy;
+        }
+    }
+
+    /**
+     * TableDetails enables us to provide tests with sufficient physical information such that they can verify that
+     * everything that _should_ be cleaned up has been cleaned up.
+     *
+     * Generally, tests will be working on a collection of tables and will want to compare against the currently
+     * persisted table names. Thus, this also provides some utility methods for mapping from a collection of
+     * TableDetails down to a set of a particular constituent part
```

Comment:
nit: hanging comment?

Reply:
That was intended to be the entire comment - I'll recheck the wording...